_posts/2017-11-16-totw-99.md

```diff
@@ -0,0 +1,239 @@
+---
+title: "Tip of the Week #99: Nonmember Interface Etiquette"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/99
+type: markdown
+order: "099"
+---
+
+Originally posted as totw/99 on 2015-06-24
+
+By a former Googler
+
+Revised 2017-10-10
+
+The interface of a C++ class is not constrained to its members or to its
+definition. When evaluating an API, we must consider definitions beyond the
+class body that can be as much a part of its interface as its public members.
+
+These external interface points include template specializations like hashers or
+traits, nonmember operator overloads (e.g. logging, relationals), and other
+canonical nonmember functions designed for use with argument-dependent lookup
+(ADL), most notably `swap`.
+
+A few of these are illustrated below for some sample class `space::Key`:
+
+```c++
```

Comment:
I think we should standardize on indent 2 spaces for tips. Any issues with that?

Reply:
namespaces aren't indented.